app/src/main/java/de/tum/in/tumcampusapp/component/tumui/roomfinder/WeekViewFragment.kt

```diff
@@ -0,0 +1,106 @@
+package de.tum.`in`.tumcampusapp.component.tumui.roomfinder
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import com.alamkanak.weekview.MonthChangeListener
+import com.alamkanak.weekview.WeekView
+import com.alamkanak.weekview.WeekViewDisplayable
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.api.app.TUMCabeClient
+import de.tum.`in`.tumcampusapp.component.tumui.calendar.WidgetCalendarItem
+import de.tum.`in`.tumcampusapp.utils.Const
+import de.tum.`in`.tumcampusapp.utils.Utils
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+import java.util.*
+
+class WeekViewFragment : Fragment(), MonthChangeListener<WidgetCalendarItem> {
+
+    private val eventsCache = HashMap<Calendar, List<WeekViewDisplayable<WidgetCalendarItem>>>()
+
+    private var roomApiCode: String? = null
+    lateinit var mWeekView: WeekView<WidgetCalendarItem>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        roomApiCode = arguments?.getString(Const.ROOM_ID)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater,
+                              container: ViewGroup?, savedInstanceState: Bundle?): View? =
+            inflater.inflate(R.layout.fragment_day_view, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        mWeekView = view.findViewById(R.id.weekView)
+        mWeekView.setMonthChangeListener(this)
+        mWeekView.goToHour(8)
+    }
+
+    override fun onMonthChange(startDate: Calendar,
+                               endDate: Calendar): List<WeekViewDisplayable<WidgetCalendarItem>> {
+        if (!isLoaded(startDate)) {
+            loadEventsInBackground(startDate, endDate)
+            return emptyList()
+        }
+
+        return eventsCache[startDate] ?: throw IllegalStateException()
+    }
+
+    private fun isLoaded(start: Calendar): Boolean {
+        return eventsCache[start] != null
+    }
+
+    private fun loadEventsInBackground(start: Calendar, end: Calendar) {
+        // Populate the week view with the events of the month to display
+        Thread {
```

Comment:
Consider using RxJava for loading the events in the background. Easy cancellation in `onDestroy()` and no `Thread().start()`.